mlflow/genai/judges/tools/registry.py

```diff
@@ -0,0 +1,124 @@
+"""
+Tool registry for MLflow GenAI judges.
+
+This module provides a registry system for managing and invoking JudgeTool instances.
+"""
+
+import json
+from typing import Any
+
+from mlflow.entities.trace import Trace
+from mlflow.exceptions import MlflowException
+from mlflow.genai.judges.tools.base import JudgeTool
+from mlflow.protos.databricks_pb2 import RESOURCE_DOES_NOT_EXIST
+from mlflow.types.llm import ToolCall
+from mlflow.utils.annotations import experimental
+
+
+@experimental(version="3.4.0")
+class JudgeToolRegistry:
+    """Registry for managing and invoking JudgeTool instances."""
+
+    def __init__(self):
+        self._tools: dict[str, JudgeTool] = {}
+
+    def register(self, tool: JudgeTool) -> None:
+        """
+        Register a judge tool in the registry.
+
+        Args:
+            tool: The JudgeTool instance to register
+        """
+        self._tools[tool.name] = tool
+
```

Comment:
Do we want to add a de-register method to this class?

Reply:
Even though this is purely backend, I'm wondering if a purger might be helpful in case a particular tool fails too many times and it can be removed as 'not applicable for use for this judge'?